linux_os/guide/system/software/sudo/sudo_custom_logfile/rule.yml

```diff
@@ -31,4 +31,5 @@ template:
     name: sudo_defaults_option
     vars:
         option: logfile
+        option_regex_suffix: '=(.*)'
```

Comment:
And could you add this new parameter in the template documentation?

Reply:
Most future proof would be to craft a regex that matches the logfile path without going further into other options.

Reply:
Added to the docs :-)

See commentary above about the general case. Would the preferred solution be `=([^,]*)` or do you have another regex in mind?

Reply:
I think lazy solution could be: `=([\w\/.]+)`. But it should work if the regex matches any character allowed in a path.
Hmm, can a filename have a comma?

Reply:
Commas are allowed.
But then, if a logfile path can have commas, I don't know how sudo itself would parse the config correctly.
Unless you could wrap the path in quotes...

Reply:
> Values may be enclosed in double quotes ("") when they contain multiple words.  Special characters may be escaped with a backslash (‘\’).

It is underspecified as to whether or not quotes remove the need for `\` backslash escpaing. I dunno. :-)

Reply:
Grab some :popcorn:, what do you think of:

> ```
> =("(?:(?:\\"|[^"])*)*"|[^"](?:(?:\\,|\\"|[^",])*))
> ```

Outer capture group for the value.

Inside its an OR:

 - Either we have quotes.
 - Or we don't have quotes. 

In the quote path, we match the start and end quote (admin must specify in the XCCDF variable). Then we match any character group that's either an escaped double quote or isn't a double quote. Repeat until we have our value.

Same for the non-double-quote path: we match any character group that's either an escaped double-quote, an escaped comma, or any non-comma non-double-quote character. Repeat until we have our value. 

---

Discussed this with @monoidmusician and he suggested something along the lines of:

> ```
> (([^",]|\\,)*|"([^"]|\\")*")*
> ```

But I had issues getting that to work (it broke on escaped commas) and came up with the above.

WDYT?

Reply:
Take another bite of :popcorn:

> ```
> =("(?:(?:\\"|[^"])*)*"|[^"](?:(?:\\,|\\"|\\ |[^", ])*))
> ```

To the non-quote path I:

 - Added space as an escapable-character
 - Added space to the non-match (for standalone space) group at the end.

Thanks @MonoidMusician for the catch.

Reply:
Ooops... Still broken. I think this one works better:

> ```
> =("(?:\\"|\\\\|[^"\\\n])*"\B|[^"](?:(?:\\,|\\"|\\ |\\\\|[^", \\\n])*)\b)
> ```

Note that this will require pulling the `\b` out of the [OVAL pattern](https://github.com/ComplianceAsCode/content/blob/master/shared/templates/sudo_defaults_option/oval.template#L16) and putting it into the default `option_regex_suffix`. 

<details>
<summary>
Test cases:
</summary>


```
Defaults logfile="/var/this_path"
Defaults logfile="/var/this path"
Defaults logfile="/var/this_path",noexec
Defaults logfile="/var/this path",noexec
Defaults logfile=/var/log,noexec
Defaults logfile=/var/log\,noexec,noexec
Defaults logfile=/var/log/\ noexec,noexec
Defaults logfile=/var/log/insane\"quotes
Defaults logfile=/var/log/insane\"quotes,noexec
Defaults logfile=/var/log no quotes
Defaults logfile=/var/log no quotes,noexec
Defaults logfile=/var/log\ with\ escaped\ spaces
Defaults logfile=/var/log\ with\ escaped\ spaces,noexec
Defaults noshm,logfile="/var/this_path"
Defaults noshm,logfile="/var/this path"
Defaults noshm,logfile="/var/this_path",noexec
Defaults noshm,logfile="/var/this path",noexec
Defaults noshm,logfile=/var/log,noexec
Defaults noshm,logfile=/var/log\,noexec,noexec
Defaults noshm,logfile=/var/log/\ noexec,noexec
Defaults noshm,logfile=/var/log/insane\"quotes
Defaults noshm,logfile=/var/log/insane\"quotes,noexec
Defaults noshm,logfile=/var/log no quotes
Defaults noshm,logfile=/var/log no quotes,noexec
Defaults noshm,logfile=/var/log\ with\ escaped\ spaces
Defaults noshm,logfile=/var/log\ with\ escaped\ spaces,noexec
```

</details>

Reply:
Regexp reviewed. Looks solid.

Reply:
Love the test set.
Any chance of adding the explanation as comment in the rule?

> Note that this will require pulling the `\b` out of the [OVAL pattern](https://github.com/ComplianceAsCode/content/blob/master/shared/templates/sudo_defaults_option/oval.template#L16) and putting it into the default `option_regex_suffix`.

Ok, this should not be a problem if incorporated into the default `option_regex_suffix`, right?

> Take another bite of popcorn

The emoji looks more like beer to me...  Was thinking who bites beer?

Reply:
> Ok, this should not be a problem if incorporated into the default `option_regex_suffix`, right?

Yep, that's exactly what I did :-) Just something to keep in mind if anyone else wishes to do something similar. 

@yuumasato Did you want me to somehow add the test set to the SSGTS? Or could I leave it out? Wouldn't I effective need a ton of different profiles to set the variable value differently? Just curious how that'd work. Or does it suffice to leave the test suite here/in the commit?

Reply:
> Did you want me to somehow add the test set to the SSGTS? Or could I leave it out? 

Thinking about, it would be good to have at least one test scenario, for sanity checking purposes.

> Wouldn't I effective need a ton of different profiles to set the variable value differently? Just curious how that'd work.

Not profiles, but individual test scenarios overwriting the variable with `# variable` Like this:
https://github.com/ComplianceAsCode/content/blob/95d892d7d058698c945115042f4901f8d5496215/linux_os/guide/system/software/sudo/sudo_add_umask/tests/0027_var_and_default_state.fail.sh#L3

> Or does it suffice to leave the test suite here/in the commit?

The bulk of tests can be in the commit.